lib/extensions/ar_taggable.rb

```diff
@@ -77,6 +77,38 @@ def tags(options = {})
       options[:ns] = Tag.get_namespace(options)
       Tag.tags(options)
     end
+
+    # defines an attribute that detects a certain tag
+    # define has_attrs, has_attrs? and attr_tags
+    # arel returns true or nil (false)
+    #   a count() > 0 or a case statement could return true, false (instead of null) but this seemed the least verbose
+    def tag_attribute(attribute_name, ns)
+      plural_attribute_name = "has_#{attribute_name.to_s.pluralize}"
+
+      virtual_attribute plural_attribute_name, :boolean, :uses => :tags, :arel => (lambda do |t|
+        ta = Tag.arel_table
+        tnga = Tagging.arel_table
+        t.grouping(
+          Arel.sql(
+            Tagging.joins(:tag).select('true')
+                   .where(ta[:name].matches("#{ns}/%", nil, true))
+                   .where(tnga[:taggable_type].eq(base_class.name).and(tnga[:taggable_id].eq(arel_table[:id])))
+                   .limit(1).to_sql
+          )
+        )
+      end)
+
+      define_method("#{attribute_name}_tags") do
+        Tag.filter_ns(tags, ns)
+      end
+
+      define_method(plural_attribute_name) do
+        has_attribute?(plural_attribute_name) ? self[plural_attribute_name] || false : Tag.filter_ns(tags, ns).size > 0
```

Comment:
Is `self[plural_attribute_name]` preferable over `self.public_send(plural_attribute_name)` ?  I've always tried to avoid `.attributes` or `.[]` because they used to create the entire attributes hash under the covers even if you didn't need it.

Reply:
Should the `Tag.filter_ns(tag, ns)` just be a call to ` .send("#{attribute_name}_tags")` to DRY it up?

Reply:
Also, is `.any?` preferable over `.size > 0`

Reply:
> Is `self[plural_attribute_name]` preferable over `self.public_send(plural_attribute_name)`?

@Fryguy pretty sure that would just cause infinite recursion on this method.

Reply:
oh duh...then I guess I was more thinking using `read_attribute(plural_attribute_name`

Reply:
I'll change `size > 0` over to `exists?` - think that is correct. sigh. So hard to remember which method to use to determine `AR#empty?` / `AR#present?`

Reply:
@NickLaMuro @Fryguy did I resolve all the requests with this comment?